components/supervisor/pkg/supervisor/top.go

```diff
@@ -10,17 +10,68 @@ import (
 	"os"
 	"strconv"
 	"strings"
+	"sync"
 	"time"
 
 	"golang.org/x/xerrors"
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/credentials/insecure"
 
 	linuxproc "github.com/c9s/goprocinfo/linux"
+	"github.com/gitpod-io/gitpod/common-go/log"
 	"github.com/gitpod-io/gitpod/supervisor/api"
 	daemonapi "github.com/gitpod-io/gitpod/ws-daemon/api"
 )
 
+type TopService struct {
+	data      *api.ResourcesStatusResponse
+	ready     chan struct{}
+	readyOnce sync.Once
+}
+
+func (t *TopService) GetTopData() *api.ResourcesStatusResponse {
+	var newData *api.ResourcesStatusResponse = t.data
+	return newData
+}
+
+// Observe starts observing the resource status
+func (t *TopService) Observe(ctx context.Context) {
+	var (
+		delay                       = 1 * time.Second
+		reconnectionDelayGrowFactor float64
```

Comment:
@andreafalzetti Maybe a dump question buy where is this initialized before being used in line 66?

Reply:
good point it should be assigned to 1.5

Reply:
good catch, updated, thanks 🤦 🙏